pf/tests/provider_create_test.go

```diff
@@ -174,3 +174,61 @@ func TestCreateWithFirstClassSecrets(t *testing.T) {
 	}`
 	testutils.Replay(t, server, testCase)
 }
+
+func TestCreateWithSchemaBasedSecrets(t *testing.T) {
+	// Ensure that resources that mark output properties as secret in the schema return them as secrets.
+	// RandomPassword is a good example. Surprisingly this test requires a Configure call first, otherwise the
+	// plubming is confused about secrets bits and retursn the wrong result. The test represents production use.
+	server := newProviderServer(t, testprovider.RandomProvider())
+	testCase := `
+	[
+	  {
+	    "method": "/pulumirpc.ResourceProvider/Configure",
+	    "request": {
+	      "args": {},
+	      "acceptSecrets": true,
+	      "acceptResources": true
+	    },
+	    "response": {
+	      "acceptSecrets": true,
+	      "supportsPreview": true,
+	      "acceptResources": true
+	    }
+	  },
+	  {
+	    "method": "/pulumirpc.ResourceProvider/Create",
+	    "request": {
+	      "urn": "urn:pulumi:dev::secret-random-yaml::random:index/randomPassword:RandomPassword::param",
+	      "properties": {
+		"length": 10
+	      }
+	    },
+	    "response": {
+	      "id": "none",
+	      "properties": {
+		"__meta": "{\"schema_version\":\"3\"}",
+		"bcryptHash": {
+		  "4dabf18193072939515e22adb298388d": "1b47061264138c4ac30d75fd1eb44270",
```

Comment:
This works apparently because convert package is looking at the Schema and doing secret-aware conversion to PropertyValue.